packages/x-charts/src/internals/plugins/utils/selectors.ts

```diff
@@ -1,11 +1,30 @@
 import { lruMemoize, createSelectorCreator, CreateSelectorFunction } from 'reselect';
+import { fastShallowCompare } from '@mui/x-internals/fastShallowCompare';
 import { ChartAnyPluginSignature, ChartState, ChartStateCacheKey } from '../models';
 
+const complainIfNoMemoize =
```

Comment:
Should we use this? I guess it makes sense. Is there any way to remove code when building for a release?

Reply:
`equalityCheck: process.env.NODE_ENV !== 'production' ? compain : fast`

Reply:
This PR is mostly for discussion on the issue, it does not depict the final solution. 😄 

I think the discussion would be more if we want to use `fastShallowCompare` directly, which would fix the problems everywhere.

Or if we want to have this `complain` function, and manually fix those when we detect them.